pkg/dns/server/name_table.go

```diff
@@ -63,7 +63,14 @@ func BuildNameTable(cfg Config) *dnsProto.NameTable {
 			if svc.Resolution == model.Passthrough && len(svc.Ports) > 0 {
 				for _, instance := range cfg.Push.ServiceEndpointsByPort(svc, svc.Ports[0].Port, nil) {
 					// empty addresses are possible here
-					if !netutil.IsValidIPAddress(instance.Address) {
+					isValidInstance := true
+					for _, addr := range instance.Addresses {
+						if !netutil.IsValidIPAddress(addr) {
+							isValidInstance = false
+							break
+						}
+					}
+					if !isValidInstance {
```

Comment:
This doesn't really make sense?

Before, we were checking to make sure we didn't have any `Address=""` case. I don't see how we can have `Addresses=[]string{""}`.

Actually this seems to NOT cover the case we used to, where we have unset addresses (`Addresses=[]string{}`).

Reply:
Later on L92 this can result in invalid response with empty address

Reply:
Hi @howardjohn, I add another assertion for checking empty `Addresses` via `if len(instance.Addresses) == 0`, and the original logic checks the empty address, besides that it also check the validation of the IP address by golang ipnet package. So I suggest to keep this.